src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestStream.cs

```diff
@@ -122,11 +144,11 @@ public override int Read(byte[] buffer, int offset, int count)
             throw new NotSupportedException();
         }
 
-        internal void EndUpload()
+        internal async Task EndUploadAsync(CancellationToken token)
```

Comment:
Sounds like WinHttp knows chunked. 
If it doesn't couldn't we overload CopyToAsync and add the last chunk there somehow?